app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -21,6 +21,36 @@ def build_config_disk_spec(vmcs)
     end
   end
 
+  def build_disk_relocate_spec(clone_options)
+    get_disks_info.map do |disk|
+      VimHash.new("VirtualDiskFlatVer2BackingInfo") do |vdbi|
+        case get_option(:disk_format)
+        when 'thin'
+          vdbi.thinProvisioned = true # VimString.new('sparse', "VirtualMachineRelocateTransformation")
+        when 'thick'
+          vdbi.thinProvisioned = false
+          vdbi.eagerlyScrub = false # then VimString.new('flat', "VirtualMachineRelocateTransformation")
+        when 'thick_eager'
+          vdbi.eagerlyScrub = true
+          vdbi.thinProvisioned = false
+        end
+      end
+    end
+  end
+
+  def get_disks_info
+    inventory_hash = source.with_provider_connection do |vim|
+      vim.virtualMachineByMor(source.ems_ref_obj)
+    end
+
+    devs = inventory_hash.fetch_path("config", "hardware", "device") || []
```

Comment:
@agrare I can move lines 42-46 here to a private helper method inside here, right?